drv/lpc55-gpio/src/main.rs

```diff
@@ -137,8 +145,150 @@ impl idl::InOrderPinsImpl for ServerImpl<'_> {
             core::ptr::write_volatile(base as *mut u32, conf);
         }
 
+        // If the GPIO pin is configured for interrupts, then
+        // interrupt configuration is done before pin configuration.
+        // INPUTMUX from UM11126:
+        // Once set up, no clocks are required for the input multiplexer to
+        // function. The system clock is needed only to write to or read from
+        // the INPUT MUX registers. Once the input multiplexer is configured,
+        // disable the clock to the INPUT MUX block in the AHBCLKCTRL register.
+        // See: Section 4.5.17 “AHB clock control 0”.
+        match pint_slot {
+            PintSlot::None => {}
+            PintSlot::Slot0
+            | PintSlot::Slot1
+            | PintSlot::Slot2
+            | PintSlot::Slot3
+            | PintSlot::Slot4
+            | PintSlot::Slot5
+            | PintSlot::Slot6
+            | PintSlot::Slot7 => {
+                // Ensure that the pin is configured as an input during
+                // PINT setup. XXX remove if not needed.
+                let (tport, tpin) = gpio_port_pin_validate(pin);
+                self.set_pin_direction(tport, tpin, Direction::Input);
+
+                // The INPUTMUX only needs to be turned on during configuration.
+                let syscon = Syscon::from(SYSCON.get_task_id());
+                syscon.enable_clock(Peripheral::Mux);
+                syscon.leave_reset(Peripheral::Mux);
+                unsafe {
+                    self.inputmux.pintsel[pint_slot as usize]
+                        .write(|w| w.bits(pin as u32));
+                }
```

Comment:
Can we use the `intpin` API which would let us wrap the unsafe a bit tighter or does that not work with our hubris design?